mobile/ios/device/ios-device-file-system.ts

```diff
@@ -42,27 +43,32 @@ export class IOSDeviceFileSystem implements Mobile.IDeviceFileSystem {
 		}).future<any>()();
 	}
 
-	public getFile(deviceFilePath: string): IFuture<void> {
-		return (() => {
-			let afcClient = this.resolveAfc();
-			let fileToRead = afcClient.open(deviceFilePath, "r");
-			let fileToWrite = this.$options.file ? this.$fs.createWriteStream(this.$options.file) : process.stdout;
-			let dataSizeToRead = 8192;
-			let size = 0;
-
-			while(true) {
-				let data = fileToRead.read(dataSizeToRead);
-				if(!data || data.length === 0) {
-					break;
-				}
-				fileToWrite.write(data);
-				size += data.length;
+	public getFile(deviceFilePath: string, outputFilePath?: string): IFuture<void> {
+		let future = new Future<void>();
+		let afcClient = this.resolveAfc();
+		let fileToRead = afcClient.open(deviceFilePath, "r");
+		let fileToWrite = outputFilePath ? this.$fs.createWriteStream(outputFilePath) : process.stdout;
+		if (outputFilePath) {
+			fileToWrite.on("close", () => {
+				future.return();
+			});
+		}
+		let dataSizeToRead = 8192;
+		let size = 0;
+		while(true) {
```

Comment:
Same as above, you can pipe the readable stream to writeable stream:
Instead of using `while`, you can just pipe the code to readable stream to `process.stdout`, for example:
```TypeScript
const future = new Future<void>();
let afcClient = this.resolveAfc();
let fileToRead = afcClient.open(deviceFilePath, "r");
let fileToWrite = outputFilePath ? this.$fs.createWriteStream(outputFilePath) : process.stdout;

fileToRead.pipe(fileToWrite);
fileToRead.on("end", () => {
	future.return();
});
fileToRead.on("error", (err: Error) => {
	future.throw(err);
});

return future.
```